main.cpp

```diff
@@ -68,16 +80,18 @@ int main(int argc, char *argv[])
 
     QObject *root = engine.rootObjects().first();
 
-    if (argc > 2 && strcmp(argv[1], "--log-level") == 0) {
-        if (argc > 4 && strcmp(argv[3], "--log-file") == 0) {
-            QMetaObject::invokeMethod(root, "enableLoggingToFile", Q_ARG(QVariant, argv[2]), Q_ARG(QVariant, argv[4]));
+
+    if (cliParser.isSet("log-level")) {
+        if (cliParser.isSet("log-file")) {
+            QMetaObject::invokeMethod(engine.rootObjects().first(), "enableLoggingToFile", Q_ARG(QVariant, cliParser.value("log-level")), Q_ARG(QVariant, cliParser.value("log-file")));
         } else {
-            QMetaObject::invokeMethod(root, "enableLogging", Q_ARG(QVariant, argv[2]));
+            QMetaObject::invokeMethod(engine.rootObjects().first(), "enableLogging", Q_ARG(QVariant, cliParser.value("log-level")));
```

Comment:
We can still use `root` here, no (also 2 more occurences)?

```suggestion
            QMetaObject::invokeMethod(root, "enableLogging", Q_ARG(QVariant, cliParser.value("log-level")));
```

Reply:
Yes, fixing (Copy paste error).